test/Microsoft.AspNetCore.Mvc.FunctionalTests/TempDataPropertyTest.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Net.Http.Headers;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.FunctionalTests
+{
+    public class TempDataPropertyTest : IClassFixture<MvcTestFixture<BasicWebSite.Startup>>
+    {
+        protected HttpClient Client { get; }
+
+        public TempDataPropertyTest(MvcTestFixture<BasicWebSite.Startup> fixture)
+        {
+            Client = fixture.Client;
+        }
+
+        [Fact]
+        public async Task TempDataPropertyAttribute_RetainsTempDataWithView()
+        {
+            // Arrange
+            var message = "Success (from Temp Data)";
+            var nameValueCollection = new List<KeyValuePair<string, string>>
+            {
+                new KeyValuePair<string, string>("FullName", "Bob"),
+                new KeyValuePair<string, string>("id", "1"),
+            };
+            var expected = $"{message} for person {nameValueCollection[0].Value} with id {nameValueCollection[1].Value}.\r\n";
```

Comment:
Are you sure `\r\n` works in Travis? Can you trim the actual response so you don't have to include this?